sound/soc/sof/virtio-miscdev.c

```diff
@@ -0,0 +1,255 @@
+// SPDX-License-Identifier: (GPL-2.0 OR BSD-3-Clause)
+/*
+ * This file is provided under a dual BSD/GPLv2 license.  When using or
+ * redistributing this file, you may do so under either license.
+ *
+ * Copyright(c) 2017 Intel Corporation. All rights reserved.
+ *
+ *  Author: Libin Yang <libin.yang@intel.com>
+ *	  Luo Xionghu <xionghu.luo@intel.com>
+ *	  Liam Girdwood <liam.r.girdwood@linux.intel.com>
+ */
+
+#include <linux/module.h>
+#include <linux/miscdevice.h>
+#include <linux/fs.h>
+#include <linux/file.h>
+#include <linux/slab.h>
+
+#include "sof-priv.h"
+#include "virtio-miscdev.h"
+#include <linux/vbs/vbs.h>
+
+/*
+ * This module registers a device node /dev/vbs_k_audio,
+ * that handle the communication between Device Model and
+ * the virtio backend service. The device model can
+ * control the backend to : set the status, set the vq account and etc.
+ * The config of the DM and VBS must be accordance.
+ */
+
+static struct virtio_miscdev *virtio_audio;
+
+static struct virtio_miscdev *get_virtio_audio(void)
```

Comment:
snd_sof_virtio_get_miscdev()

Reply:
Libin said he will push BE part himself, so I will remove the BE related in this patch, and resend the FE part only